src/groupByKey.ts

```diff
@@ -2,6 +2,21 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T extends object>(
+  items: T[],
+  key: keyof T,
+): { [k: string]: T[] } {
+  const result: GroupsMap<T> = {};
+
+  for (const item of items) {
+    const value = item[key] as string;
```

Comment:
The assumption that `item[key]` is always a string may not hold true for all objects. Consider converting `item[key]` to a string using `String(item[key])` to ensure compatibility with all types of keys.